cloud-controller-manager/do/loadbalancers.go

```diff
@@ -439,60 +446,78 @@ func buildForwardingRules(service *v1.Service) ([]godo.ForwardingRule, error) {
 	certificateID := getCertificateID(service)
 	tlsPassThrough := getTLSPassThrough(service)
 
-	if (certificateID != "" || tlsPassThrough) && len(tlsPorts) == 0 {
+	if len(tlsPorts) == 0 && (protocol == protocolHTTPS || certificateID != "" || tlsPassThrough) {
 		tlsPorts = append(tlsPorts, 443)
 	}
 
+	tlsPortMap := map[int32]bool{}
+	for _, port := range tlsPorts {
+		tlsPortMap[int32(port)] = true
+	}
+
 	// If using sticky sessions and no (or tcp) protocol was specified,
 	// default to HTTP.
 	stickySessionsType := getStickySessionsType(service)
-	if stickySessionsType == stickySessionsTypeCookies && protocol == "tcp" {
-		protocol = "http"
+	if stickySessionsType == stickySessionsTypeCookies && protocol == protocolTCP {
+		protocol = protocolHTTP
 	}
 
-	if len(tlsPorts) > 0 {
-		if certificateID == "" && !tlsPassThrough {
-			return nil, errors.New("must set certificate id or enable tls pass through")
+	for _, port := range service.Spec.Ports {
+		// We use https for TLS, so set it explicitly if necessary.
+		if tlsPortMap[port.Port] {
+			protocol = protocolHTTPS
 		}
 
-		if certificateID != "" && tlsPassThrough {
-			return nil, errors.New("either certificate id should be set or tls pass through enabled, not both")
+		forwardingRule, err := buildForwardingRule(service, &port, protocol, certificateID, tlsPassThrough)
+		if err != nil {
+			return nil, err
 		}
+		forwardingRules = append(forwardingRules, *forwardingRule)
 	}
 
-	var forwardingRules []godo.ForwardingRule
-	for _, port := range service.Spec.Ports {
-		var forwardingRule godo.ForwardingRule
-		if port.Protocol != "TCP" {
-			return nil, fmt.Errorf("only TCP protocol is supported, got: %q", port.Protocol)
-		}
+	return forwardingRules, nil
+}
 
-		forwardingRule.EntryProtocol = protocol
-		forwardingRule.TargetProtocol = protocol
+func buildForwardingRule(service *v1.Service, port *v1.ServicePort, protocol, certificateID string, tlsPassThrough bool) (*godo.ForwardingRule, error) {
+	var forwardingRule godo.ForwardingRule
 
-		forwardingRule.EntryPort = int(port.Port)
-		forwardingRule.TargetPort = int(port.NodePort)
+	if port.Protocol != "TCP" {
```

Comment:
```suggestion
	if port.Protocol != protocolTCP {
```

Reply:
This would have to be something like `if strings.ToLower(port.Protocol) != protocolTCP`, WDYT?

Reply:
Ah true, missed the case sensitivity. The constants are also not really meant to compare for service port protocols, so please ignore this one.